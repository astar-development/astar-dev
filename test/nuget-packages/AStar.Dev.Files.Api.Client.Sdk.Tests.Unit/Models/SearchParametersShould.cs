using AStar.Dev.Files.Api.Client.SDK.Models;

namespace AStar.Dev.Files.Api.Client.Sdk.Models;

public sealed class SearchParametersShould
{
    [Fact]
    public void ReturnTheExpectedToString() =>
        new SearchParameters
            {
                CurrentPage              = 1,
                ExcludedViewSettings     = new () { ExcludeViewed = true, ExcludeViewedPeriodInDays = 12 },
                IncludeMarkedForDeletion = true,
                IncludeSoftDeleted       = true,
                ItemsPerPage             = 12,
                MaximumSizeOfImage       = 123,
                MaximumSizeOfThumbnail   = 456,
                Recursive                = true,
                SearchFolder             = "MockSearchFolder",
                SearchText               = "MockSearchText",
                SearchType               = SearchType.All,
                SortOrder                = SortOrder.NameAscending
            }
            .ToString()
            .ShouldMatchApproved();

    [Fact]
    public void ReturnTheExpectedToQueryString() =>
        new SearchParameters
            {
                CurrentPage              = 1,
                ExcludedViewSettings     = new () { ExcludeViewed = true, ExcludeViewedPeriodInDays = 12 },
                IncludeMarkedForDeletion = true,
                IncludeSoftDeleted       = true,
                ItemsPerPage             = 12,
                MaximumSizeOfImage       = 123,
                MaximumSizeOfThumbnail   = 456,
                Recursive                = true,
                SearchFolder             = "MockSearchFolder",
                SearchText               = "MockSearchText",
                SearchType               = SearchType.All,
                SortOrder                = SortOrder.NameAscending
            }
            .ToQueryString()
            .ShouldMatchApproved();
}
