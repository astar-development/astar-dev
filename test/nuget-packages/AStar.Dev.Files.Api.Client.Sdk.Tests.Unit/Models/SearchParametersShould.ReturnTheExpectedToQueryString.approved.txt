SearchFolder=MockSearchFolder&CurrentPage=1&ItemsPerPage=12&SearchType=All&Recursive=True&IncludeSoftDeleted=True&IncludeMarkedForDeletion=True&ExcludeViewed=True&SortOrder=NameAscending&MaximumSizeOfThumbnail=456&MaximumSizeOfImage=123&ExcludeViewedPeriodInDays=12&SearchText=MockSearchText